rancher/service.go

```diff
@@ -349,10 +349,7 @@ func (r *RancherService) getLinks() (map[Link]string, error) {
 		name = strings.TrimSpace(name)
 		alias = strings.TrimSpace(alias)
 
-		linkedService, err := r.FindExisting(name)
-		if err != nil {
-			return nil, err
-		}
+		linkedService, _ := r.FindExisting(name)
```

Comment:
I think the right fix is to fix `FindExisting` method so that if the external stack/service is not present in current environment, return a nil service and error. In here you are also ignoring other errors which might cause a problem.